### PR TITLE
docs: Add ADR for creation of Trust Framework Adoption repository

### PR DESCRIPTION
**Pull request by bscholtes1A**
## What this PR changes/adds

Add ADR for creation of Gaia-X Bundle repository.

## Why it does that

Related to EDC adoption in Gaia-X.

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] assigned appropriate label? (exclude from changelog with label `no-changelog`)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: docs/developer/decision-records/2022-10-20-gaiax-bundle-repository/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+to enable trust and interoperability. This implies some constraints on how the participant's data exchange components are deployed and operated in such Gaia-X dataspace, e.g. identity enforcement, access control...
+
+In order to ease adoption of the EDC into Gaia-X dataspaces, it was decided to create a dedicated _Gaia-X Bundle_ repository under the EDC project, which will
+contain all extensions and relevant documentation for deploying and using the EDC with respect of the Gaia-X Trust Framework.
+These extensions will be packaged and published in the same way as the other EDC components, making it easy to build a Gaia-X compliant version of the EDC.
+
+> ⚠️ The EDC project does not aim at supporting all trust frameworks that might be developed at a later stage, and that might be or not based on Gaia-X.
+> Thus, only the present Gaia-X Bundle repository will be created and maintained under the EDC project. Any other future framework would have to
+> develop and maintain its own set of dedicated extensions.
+
```
Comment (mspiekermann) [inline docs/developer/decision-records/2022-10-20-gaiax-bundle-repository/README.md:24]: The EDC project does not aim to support all future trust frameworks in a similar form and make them available as bundles as comparable artifacts. Compliance with Gaia-X was stated as a goal in the proposal on the part of the EDC project and takes a separate role in the consideration. 
Thus, only the present Gaia-X Bundle repository will be created and maintained under the EDC project, which can be used as a blueprint for other configurations. Any other future framework would have to develop, maintain and publish its own extensions.

**Review by mspiekermann — APPROVED** [review]

**Review by juliapampus — APPROVED** [review]
@bscholtes1A Can you add a link to this file to the list in `decision-records/README.md`?

File: docs/developer/decision-records/2022-10-20-gaiax-bundle-repository/README.md (hunk 1 above)
Comment (jimmarino) [reply docs/developer/decision-records/2022-10-20-gaiax-bundle-repository/README.md:24]: +1 on the concept. One thing we should think about is the best place for "code" extensions vs. BOMs. For example, let's assume we have:

1. A policy extension that includes a policy function to evaluate a Gaia-X self-description
2. An Identity Hub extension that handles the storage and rendering of Gaia-X self-description
3. Two Maven BOMs that define launcher configurations. The first configuration is for a connector that includes the policy extension. The second is for an identity hub  that includes the storage/rendering extension

I think those extensions should be placed in different locations:

1. The policy extension should go in the base EDC repo
2. The Identity Hub extension should go into the  IH repo
3. The BOM should be included as Gradle projects in the new "bundle" repository

Do you agree, and can you add this to the ADR?

Comment (jimmarino) [reply docs/developer/decision-records/2022-10-20-gaiax-bundle-repository/README.md:24]: Following up, if you want to leave the policy module in the new (Gaia-X) repo, I'm fine with that or the other way (in core)

**Comment by bscholtes1A** [comment]
> @bscholtes1A Can you add a link to this file to the list in `decision-records/README.md`?

done

**Review by mspiekermann — APPROVED** [review]

**Pull request merged**